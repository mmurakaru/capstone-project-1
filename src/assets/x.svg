<?xml version="1.0" encoding="utf-8"?>
<svg version="1.1" id="Ebene_1" xmlns="http://www.w3.org/2000/svg" xmlns:xlink="http://www.w3.org/1999/xlink" x="0px" y="0px"
	 viewBox="0 0 600 600" style="enable-background:new 0 0 600 600;" xml:space="preserve">
<path d="M454.3,575.45L295.64,416.88L145.63,567.1c-32.26,32.3-84.6,32.34-116.91,0.07
	c-32.3-32.26-32.34-84.6-0.07-116.91l0,0l150.22-150.25L28.65,149.8C-3.61,117.5-3.58,65.16,28.73,32.89s84.64-32.23,116.91,0.07
	l150.01,150.22L454.3,24.64c32.29-32.25,84.62-32.22,116.88,0.07c32.25,32.29,32.22,84.62-0.07,116.88L412.53,300.16l158.57,158.57
	c32.39,32.15,32.58,84.46,0.43,116.85c-32.15,32.39-84.46,32.58-116.85,0.43c-0.14-0.14-0.28-0.28-0.42-0.42L454.3,575.45z"/>
</svg>