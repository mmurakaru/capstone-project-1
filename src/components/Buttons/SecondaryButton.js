import styled from 'styled-components/macro'
import PropTypes from 'prop-types'

SecondaryButton.propTypes = {
  disabled: PropTypes.bool,
  handleClick: PropTypes.func,
  children: PropTypes.object.isRequired,
}

export default function SecondaryButton({ handleClick, children }) {
  return (
    <ButtonStyled onClick={handleClick} data-testid="button">
      {children}
    </ButtonStyled>
  )
}

const ButtonStyled = styled.button`
  background-color: var(--primary-dark);
  border: none;
  fill: var(--white);
  font-size: 0.8em;
  padding: 10px;
  height: 100%;
  width: 36px;
`
