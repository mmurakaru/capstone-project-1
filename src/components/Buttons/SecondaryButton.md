Regular Button with icon

```jsx
<SecondaryButton handleClick={() => {}}>
  <svg src={'../../assets/x.svg'} style={{ width: '46px', height: '46px' }} />
</SecondaryButton>
```
