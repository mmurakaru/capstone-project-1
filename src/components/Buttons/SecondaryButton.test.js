import { render } from '@testing-library/react'
import user from '@testing-library/user-event'
import SecondaryButton from './SecondaryButton'
import { ReactComponent as DeleteIcon } from '../../assets/x.svg'

describe('SecondaryButton', () => {
  it('renders correctly', () => {
    const { container } = render(
      <SecondaryButton handleClick={() => {}}>
        <DeleteIcon />
      </SecondaryButton>
    )
    expect(container.firstChild).toMatchSnapshot()
  })
  it('calls handleClick with the correct id', () => {
    const handleClickMock = jest.fn()
    const { getByRole } = render(
      <SecondaryButton handleClick={() => handleClickMock('a')}>
        <DeleteIcon />
      </SecondaryButton>
    )
    const button = getByRole('button')
    user.click(button)

    expect(handleClickMock).toHaveBeenCalledWith('a')
  })
})
