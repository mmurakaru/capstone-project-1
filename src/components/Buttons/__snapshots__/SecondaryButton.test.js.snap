// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SecondaryButton renders correctly 1`] = `
.c0 {
  background-color: var(--primary-dark);
  border: none;
  fill: var(--white);
  font-size: 0.8em;
  padding: 10px;
  height: 100%;
  width: 36px;
}

<button
  class="c0"
  data-testid="button"
>
  <svg>
    x.svg
  </svg>
</button>
`;
